pkg/driver/node/mounter/pod_mounter.go

```diff
@@ -384,3 +432,48 @@ func (pm *PodMounter) volumeNameFromTargetPath(target string) (string, error) {
 func (pm *PodMounter) helpMessageForGettingMountpointLogs(pod *corev1.Pod) string {
 	return fmt.Sprintf("You can see Mountpoint logs by running: `kubectl logs -n %s %s`. If the Mountpoint Pod already restarted, you can also pass `--previous` to get logs from the previous run.", pod.Namespace, pod.Name)
 }
+
+func (pm *PodMounter) getS3PodAttachmentWithRetry(ctx context.Context, volumeName string, credentialCtx credentialprovider.ProvideContext, fsGroup, pvMountOptions string) (*crdv1.MountpointS3PodAttachment, string, error) {
+	fieldFilters := client.MatchingFields{
+		crdv1.FieldNodeName:             os.Getenv("CSI_NODE_NAME"), // TODO
+		crdv1.FieldPersistentVolumeName: volumeName,
+		crdv1.FieldVolumeID:             credentialCtx.VolumeID,
+		crdv1.FieldMountOptions:         pvMountOptions,
+		crdv1.FieldWorkloadFSGroup:      fsGroup,
+		crdv1.FieldAuthenticationSource: credentialCtx.AuthenticationSource,
+	}
+	if credentialCtx.AuthenticationSource == credentialprovider.AuthenticationSourcePod {
+		fieldFilters[crdv1.FieldWorkloadNamespace] = credentialCtx.PodNamespace
+		fieldFilters[crdv1.FieldWorkloadServiceAccountName] = credentialCtx.ServiceAccountName
```

Comment:
Is it intentional that this one doesn't include role ARN? Ideally, this logic should be extracted into a common module and re-used between the node and the controller